### PR TITLE
Add Unclassified error type

### PR DESCRIPTION
**Pull request by prankulmahajan**

### CODE REVIEW

**Comment by prankulmahajan** [comment]
Added errors in https://github.com/IBM/ibmcloud-volume-interface/pull/14

**Comment by prankulmahajan** [comment]
After changing storage-secret-store to point towards incorrect exchange point URL we get output,

```
prankul@Prankuls-MacBook-Pro ibm-vpc-block-csi-driver % kubectl describe pvc csi-block-pvc-good
Name:          csi-block-pvc-good
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Used By:       <none>
Events:
  Type     Reason                Age                   From                                                                                      Message
  ----     ------                ----                  ----                                                                                      -------
  Normal   Provisioning          75s (x2 over 3m13s)   vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_a96499ce-d33e-4723-9133-f262c29812e8  External provisioner is provisioning volume for claim "default/csi-block-pvc-good"
  Warning  ProvisioningFailed    75s                   vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_a96499ce-d33e-4723-9133-f262c29812e8  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 347bbc49-e70c-412c-9d78-618d24b60388 , Code: InternalError, Description: Internal error occurred, BackendError: IAM token exchange request failed: Incorrect exchange endpoint, Action: Please check 'BackendError' tag for more details}
  Normal   ExternalProvisioning  14s (x13 over 3m13s)  persistentvolume-controller                                                               waiting for a volume to be created, either by external provisioner "vpc.block.csi.ibm.io" or manually created by system administrator

```

**Comment by prankulmahajan** [comment]
This is based on the changes made in https://github.com/IBM/ibmcloud-volume-interface/pull/14

Have created new UserError of type "EndpointNotReachable". After testing got the following output,

```
prankul@Prankuls-MacBook-Pro ibm-vpc-block-csi-driver % kubectl describe pvc csi-block-pvc-good           
Name:          csi-block-pvc-good
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Used By:       <none>
Events:
  Type     Reason                Age                From                                                                                      Message
  ----     ------                ----               ----                                                                                      -------
  Warning  ProvisioningFailed    47s                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: b1c148a8-080f-4e91-b19c-de8569346136 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    47s                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 652e3f81-73e1-4cef-b65f-4fa900655da1 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    46s                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: ad6cf11d-4b80-4f1d-bb5f-0f1a0907fe33 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    42s                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 661b0e41-4bcb-4b24-80f4-203b0b40a4ac , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    34s                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: be1edf80-77f7-4e15-979e-bc98d7336a1d , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}
  Normal   Provisioning          18s (x6 over 47s)  vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  External provisioner is provisioning volume for claim "default/csi-block-pvc-good"
  Warning  ProvisioningFailed    18s                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: d19457a3-570d-484d-a568-6c45e02c897e , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}
  Normal   ExternalProvisioning  7s (x4 over 47s)   persistentvolume-controller                                                               waiting for a volume to be created, either by external provisioner "vpc.block.csi.ibm.io" or manually created by system administrator

```

Error:
`vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_fee25e12-9b91-421f-b88b-69c864b82082  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: b1c148a8-080f-4e91-b19c-de8569346136 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://priv.containers.test.cloud.ibm.com/v1/iam/apikey": dial tcp: lookup priv.containers.test.cloud.ibm.com on 172.21.0.10:53: no such host, RC:500}, Action: Please check 'BackendError' tag for more details}`

**Comment by prankulmahajan** [comment]
Reduced the timeout, to replicate the other scenario, got the following results,

```
prankul@Prankuls-MacBook-Pro ibm-vpc-block-csi-driver % kubectl describe pvc csii-block-pvc-good                               
Name:          csii-block-pvc-good
Namespace:     default
StorageClass:  ibmc-vpc-block-5iops-tier
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: vpc.block.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Used By:       <none>
Events:
  Type     Reason                Age               From                                                                                      Message
  ----     ------                ----              ----                                                                                      -------
  Warning  ProvisioningFailed    2m                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: ea29b6ee-8534-497d-8efe-5fa862a5980f , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    2m                vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 6f85197f-691f-480c-9547-a47d616b0a3a , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    119s              vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 1de76d08-f45b-4566-b3de-2968340bf6f6 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    115s              vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 2231fcea-1e86-47d0-b2fe-9d5a4cb71e32 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    107s              vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: fb3177c2-287d-4287-8c3a-6aaf4d294f4e , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Warning  ProvisioningFailed    91s               vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: bc5df419-eddf-4053-b660-639b97678ef4 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Normal   Provisioning          59s (x7 over 2m)  vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  External provisioner is provisioning volume for claim "default/csii-block-pvc-good"
  Warning  ProvisioningFailed    59s               vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 3c544aa2-6b1a-4238-90de-9c8bd89d3cc9 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}
  Normal   ExternalProvisioning  9s (x10 over 2m)  persistentvolume-controller                                                               waiting for a volume to be created, either by external provisioner "vpc.block.csi.ibm.io" or manually created by system administrator
```

Error,
`vpc.block.csi.ibm.io_ibm-vpc-block-csi-controller-0_42b48f53-4119-4ce4-a8f5-a2b4d6afe61c  failed to provision volume with StorageClass "ibmc-vpc-block-5iops-tier": rpc error: code = Internal desc = {RequestID: 3c544aa2-6b1a-4238-90de-9c8bd89d3cc9 , Code: InternalError, Description: Internal error occurred, BackendError: {Code:EndpointNotReachable, Type:FailedAccessToken, Description:IAM_TOKEN exchange request failed., BackendError:Post "https://private.containers.test.cloud.ibm.com/v1/iam/apikey": context deadline exceeded (Client.Timeout exceeded while awaiting headers), RC:500}, Action: Please check 'BackendError' tag for more details}`

File: common/messages/messages_en.go [hunk 1]
```diff
@@ -30,6 +30,20 @@ var messagesEn = map[string]util.Message{
...
+	"EndpointNotReachable": {
+		Code:        "EndpointNotReachable",
+		Description: "Indicates IAM TOKEN exchange request failed.",
+		Type:        util.FailedAccessToken,
+		RC:          500,
+		Action:      "Verify the token_exchange_endpoint.",
+	},
+	"Timeout": {
+		Code:        "Timeout",
+		Description: "Timeout reaching to exchnage endpoint url",
```
Comment (sameshai) [inline common/messages/messages_en.go:42]: typo in exchange

File: common/messages/messages_en.go [hunk 2]
```diff
@@ -30,6 +30,20 @@ var messagesEn = map[string]util.Message{
 		RC:          400,
 		Action:      "Verify that you entered the correct IBM Cloud user name and password. If the error persists, the authentication service might be unavailable. Wait a few minutes and try again. ",
 	},
+	"EndpointNotReachable": {
+		Code:        "EndpointNotReachable",
+		Description: "Indicates IAM TOKEN exchange request failed.",
```
Comment (sameshai) [inline common/messages/messages_en.go:35]: For description can we say "IAM Token exchange request failed. Endpoint is unreachable"

File: common/messages/messages_en.go (hunk 1 above)
Comment (sameshai) [reply common/messages/messages_en.go:42]: For description can we say "IAM Token exchange request failed. Timeout reaching to endpoint"

File: common/messages/messages_en.go [hunk 3]
```diff
@@ -30,6 +30,20 @@ var messagesEn = map[string]util.Message{
...
+		Type:        util.FailedAccessToken,
+		RC:          500,
+		Action:      "Verify the token_exchange_endpoint.",
```
[the remark this answers is not in the document]
Comment (derekpoindexter) [reply common/messages/messages_en.go:38]: Verify that iks_token_exchange_endpoint_private_url is reachable from the cluster. You can find this url by running `kubectl get secret storage-secret-storage -n kube-system`.

**Review by arahamad — APPROVED** [review]
lgtm

**Review by sameshai — APPROVED** [review]

**Pull request merged**